modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/ConfigParserTestCase.java

```diff
@@ -111,6 +112,19 @@ public void sameInstanceForEquivalentEvent() throws Exception
     }
 
     @Test
+    public void sameInstanceConfigDifferentExpressionValuesForDifferentEvents() throws Exception
+    {
+        MuleEvent event = getHeisenbergEvent();
```

Comment:
this is not the correct way of testing this. What you should do is to call lookupHeisenberg twice, each with an event which produce differnet values from the config. Then, you should extract the value from the config itself and asset that you got the same config twice